src/main/scala/gestalt/Symbols.scala

```diff
@@ -15,5 +15,8 @@ trait Symbols { this: Toolbox =>
 
     /** type of a member with respect to a prefix */
     def asSeenFrom(mem: Symbol, prefix: Type): Type
+
+    /** subst symbols in tree */
+    def subst(tree: tpd.Tree)(from: List[Symbol], to: List[Symbol]): tpd.Tree
```

Comment:
Maybe it would be possible to reframe the problem, so that the typed/untyped mix is no longer required? E.g. what if `unlift` was a macro?